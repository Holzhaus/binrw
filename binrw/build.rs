use std::env;

fn main() {
    println!("cargo:rerun-if-env-changed=BINTRACE_LINK_SEARCH");
    println!("cargo:rerun-if-env-changed=BINTRACE_LINK_LIB");
    println!("cargo:rerun-if-changed=build.rs");

    if cfg!(bintrace) {
        if let Ok(link_search) = env::var("BINTRACE_LINK_SEARCH") {
            println!("cargo:rustc-link-search={}", link_search)
        }

        if let Ok(link_lib) = env::var("BINTRACE_LINK_LIB") {
            println!("cargo:rustc-link-lib={}", link_lib);
        } else {
            println!("cargo:warning=No bintrace backend provided");
        }
    }
}
