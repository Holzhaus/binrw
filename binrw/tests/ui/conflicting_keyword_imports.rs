#![feature(generic_associated_types)]
use binrw::BinRead;

#[derive(BinRead)]
#[br(import(a: i32), import_raw(args: (i32, )))]
struct Foo;

fn main() {}
