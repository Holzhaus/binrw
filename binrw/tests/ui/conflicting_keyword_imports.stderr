error: conflicting import keyword
 --> $DIR/conflicting_keyword_imports.rs:5:22
  |
5 | #[br(import(a: i32), import_raw(args: (i32, )))]
  |                      ^^^^^^^^^^
